backend/uclapi/oauth/app_helpers.py

```diff
@@ -0,0 +1,22 @@
+from binascii import hexlify
+import os
+
+
+def generate_user_token():

```

Comment:
Here you are skipping 4 letters in your key (so you use only 56 out of 60).

While I think it does not matter, the reason is quite subtle and might not be visible to the user directly.

One way how to write this function would be:
```
import os 
import textwrap


def generate_user_token():
    key = hexlify(os.urandom(30)).decode()
    dashed = '-'.join(textwrap.wrap(key, 15))
    final ="uclapi-user-" + dashed

    return final
```

Then you can observe that you have 60 hex characters instead of 56.